framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+      @ In, myInput, list, the inputs (list) to start from to generate the new
+        one
+      @ In, samplerType, string, is the type of sampler that is calling to
+        generate a new input
+      @ In,  jobHandler, JobHandler instance, the global job handler instance
+      @ In, **kwargs, dict,  is a dictionary that contains the information
+        coming from the sampler, a mandatory key is the sampledVars' that
+        contains a dictionary {'name variable':value}
+      @ Out, None
+    """
+    for mm in utils.returnImportModuleString(jobHandler):
+      if mm not in self.mods:
+        self.mods.append(mm)
+
+    prefix = kwargs['prefix']
+    self.tempOutputs['uncollectedJobIds'].append(prefix)
+    self.counter = int(prefix)
+
+    if not self.romConverged:
+      trainingSize = self.checkTrainingSize()
+      if trainingSize == self.romTrainStartSize:
+        self.trainRom(samplerType, kwargs)
+      elif trainingSize > self.romTrainStartSize and (trainingSize-self.romTrainStartSize)%self.romTrainStepSize == 0 and trainingSize <= self.romTrainMaxSize:
+        self.trainRom(samplerType, kwargs)
+      elif trainingSize > self.romTrainMaxSize:
+        self.raiseAnError(IOError, "Maximum training size is reached, but ROMs are still not converged!")
+      self.romConverged = self.checkRomConvergence()
+    if self.romConverged:
+      self.romValid = self.checkRomValidity(kwargs)
+      if not self.romValid:
+        self.romConverged = False
+
+    ## Ensemble models need access to the job handler, so let's stuff it in our
+    ## catch all kwargs where evaluateSample can pick it up, not great, but
+    ## will suffice until we can better redesign this whole process.
+    kwargs['jobHandler'] = jobHandler
+
+    ## This may look a little weird, but due to how the parallel python library
+    ## works, we are unable to pass a member function as a job because the
+    ## pp library loses track of what self is, so instead we call it from the
+    ## class and pass self in as the first parameter
+    jobHandler.addClientJob((self, myInput, samplerType, kwargs), self.__class__.evaluateSample, prefix, kwargs)
+
+  def evaluateSample(self, myInput, samplerType, kwargs):
+    """
+      This will evaluate an individual sample on this model. Note, parameters
+      are needed by createNewInput and thus descriptions are copied from there.
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+        a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, returnValue, dict, This holds the output information of the evaluated sample.
+    """
+    self.raiseADebug("Evaluate Sample")
+    jobHandler = kwargs.pop('jobHandler')
```

Comment:
please, see my PR (the one on raven running raven) ...This will make fail the EnsembleModel in case we ask to rrerun failed runs

Reply:
Changed.